provider/rfc2136/rfc2136.go

```diff
@@ -98,7 +98,7 @@ func NewRfc2136Provider(host string, port int, zoneName string, insecure bool, k
 		gssTsig:      gssTsig,
 		krb5Username: krb5Username,
 		krb5Password: krb5Password,
-		krb5Realm:    strings.ToUpper(zoneName),
+		krb5Realm:    strings.ToUpper(krb5Realm),
```

Comment:
This without any verification means that if the flag for the realm is not specified, `krb5Realm` would be empty string. This doesn't seem like a good idea to me and I would rather verify that the parameters to this function are as expected and returning an error otherwise.

Reply:
Good catch, I added realm flag validation and tests.